### PR TITLE
fix: use recently-added ConnectRPC20 compat variant

### PR DESCRIPTION
**Pull request by johnstcn**
Use the compat method added in https://github.com/coder/coder/pull/13778 
This should allow using the newer endpoint with Coder version 2.9.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,11 +4,11 @@ go 1.22.4
 
 // There are a few minor changes we make to Tailscale that we're slowly upstreaming. Compare here:
 // https://github.com/tailscale/tailscale/compare/main...coder:tailscale:main
-replace tailscale.com => github.com/coder/tailscale v1.1.1-0.20240530071520-1ac63d3a4ee3
+replace tailscale.com => github.com/coder/tailscale v1.1.1-0.20240702054557-aa558fbe5374
 
 require (
 	cdr.dev/slog v1.6.2-0.20240126064726-20367d4aede6
-	github.com/coder/coder/v2 v2.12.0
+	github.com/coder/coder/v2 v2.10.1-0.20240703121105-f6639b788f7b
```
Comment (johnstcn) [inline go.mod:11]: https://github.com/coder/coder/commit/f6639b788f7b0ad4b8f64dacc3209a6d3fb33c71

File: go.mod [hunk 2]
```diff
@@ -4,11 +4,11 @@ go 1.22.4
 
 // There are a few minor changes we make to Tailscale that we're slowly upstreaming. Compare here:
 // https://github.com/tailscale/tailscale/compare/main...coder:tailscale:main
-replace tailscale.com => github.com/coder/tailscale v1.1.1-0.20240530071520-1ac63d3a4ee3
+replace tailscale.com => github.com/coder/tailscale v1.1.1-0.20240702054557-aa558fbe5374
```
Comment (johnstcn) [inline go.mod:7]: copied from coder/coder go.mod

File: buildlog/coder.go [hunk 3]
```diff
@@ -105,11 +105,12 @@ func newAgentClientV2(ctx context.Context, logger slog.Logger, client *agentsdk.
 	var conn drpc.Conn
 	var err error
 	for r := retry.New(10*time.Millisecond, time.Second); r.Wait(ctx); {
-		conn, err = client.ConnectRPC(ctx)
+		c, err := client.ConnectRPC20(ctx)
 		if err != nil {
 			logger.Error(ctx, "connect err", slog.Error(err))
 			continue
 		}
+		conn = c.DRPCConn()
```
Comment (spikecurtis) [inline buildlog/coder.go:113]: `client.ConnectRPC20` returns the equivalent to the `arpc` below.  Grabbing the underlying `Conn` and then recreating the RPC client isn't necessary and breaks the type-checking that should prevent you from using RPCs that aren't present in v2.0 of the Agent API

**Review by spikecurtis — APPROVED** [review]
c.f. inline comment; however, I don't need to review again after it is dealt with.

**Pull request closed without merging**